File: src/api/dotnet/RealSort.cs
/*++
Copyright (c) 2012 Microsoft Corporation

Module Name:

    RealSort.cs

Abstract:

    Z3 Managed API: Real Sorts

Author:

    Christoph Wintersteiger (cwinter) 2012-11-23

Notes:
    
--*/

using System.Diagnostics;
using System;

namespace Microsoft.Z3
{
    /// <summary>
    /// A real sort
    /// </summary>
    public class RealSort : ArithSort
    {
        #region Internal
        internal RealSort(Context ctx, IntPtr obj)
            : base(ctx, obj)
        {
            Debug.Assert(ctx != null);
        }
        internal RealSort(Context ctx)
            : base(ctx, Native.Z3_mk_real_sort(ctx.nCtx))
        {
            Debug.Assert(ctx != null);
        }
        #endregion
    }
}

File: src/api/java/RealSort.java
/**
Copyright (c) 2012-2014 Microsoft Corporation
   
Module Name:

    RealSort.java

Abstract:

Author:

    @author Christoph Wintersteiger (cwinter) 2012-03-15

Notes:
    
**/ 

package com.microsoft.z3;

/**
 * A real sort
 **/
public class RealSort extends ArithSort
{
    RealSort(Context ctx, long obj) throws Z3Exception
    {
        super(ctx, obj);
    }

    RealSort(Context ctx) throws Z3Exception
    {
        super(ctx, Native.mkRealSort(ctx.nCtx()));
    }
}
